### PR TITLE
TechDocs: Replace S3 read-store-upload loop with file streams

### PR DESCRIPTION
**Pull request by lowjoel**
This no longer buffers file contents in memory.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4708-jk5ncyforz79e3mpyghtgpvfoppogizq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60387044fcb22d8df7613765

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4708-jk5ncyforz79e3mpyghtgpvfoppogizq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60387044fcb22d8df7613765

File: packages/techdocs-common/src/stages/publish/awsS3.ts [hunk 1]
```diff
@@ -167,12 +167,12 @@ export class AwsS3Publish implements PublisherBase {
         const entityRootDir = `${entity.metadata.namespace}/${entity.kind}/${entity.metadata.name}`;
         const destination = `${entityRootDir}/${relativeFilePathPosix}`; // S3 Bucket file relative path
 
-        const fileContent = await fs.readFile(filePath, 'utf8');
+        const fileStream = fs.createReadStream(filePath);
```
Comment (freben) [inline packages/techdocs-common/src/stages/publish/awsS3.ts:171]: Thanks for this!

While we're doing this change, could you move the createReadStream into the limiter below? In the interest of not running out of inodes or something similarly esoteric, we probably don't want to create an unbounded number of simultaneous streams that we don't even intend to read from right away.

Comment (lowjoel) [reply packages/techdocs-common/src/stages/publish/awsS3.ts:171]: Damn, that's a good point. Let me change.

Comment (lowjoel) [reply packages/techdocs-common/src/stages/publish/awsS3.ts:173]: @freben Updated, please have a look.

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4708-jk5ncyforz79e3mpyghtgpvfoppogizq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60387044fcb22d8df7613765

**Review by freben — APPROVED** [review]

**Comment by freben** [comment]
@OrkoHunter it doesn't seem like this landed on your board.

**Review by OrkoHunter — APPROVED** [review]
Thanks! 💯

**Comment by OrkoHunter** [comment]
Yeah @freben - the project board workflow doesn't work with PRs created from forks since they are missing the GitHub token. Usually I add them manually, but this is a small one - which I'll be merging just now.

**Pull request merged**